File: pset2/caesar.c
/*encryption program which takes an int (key) from the command line, prompts
the user for a secret message (aka plaintext) and returns the ciphertext*/


#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>


int main (int argc, string argv[])
{
    /*if user input is anything other than 2 arguments, print an error
    and return a 1 which indicates an issue and exits the program*/
    if (argc != 2) {
        printf("You must provide exactly 2 command line arguments -\
        and remember that the name of the program counts as an argument.\n");
        return 1;
    }
    
    //else if the user inputs exactly 2 command line arguments, run this encryption program
    else {
    
        //change argv from a string to an int (in order to perform mathmatical calculations per ASCII)
        int k = atoi(argv[1]);
        
        //prompt the user for the secret message (aka plaintext)
        printf("plaintext: ");
        string p = get_string();
        
        //print "ciphertext" followed by the actual ciphertext which is created using the remaining code
        printf("ciphertext: ");
        
        //iterate through each char in the plaintext
        for (int i = 0, n = strlen(p); i < n; i++) {
            
            //rotate alphabetical characters by k positions: caps stay caps,
            //lower stay lower, symbols remain unchanged
            if isalpha(p[i]) {
                
                //uppercase chars 
                if isupper(p[i]) {
                    p[i] -= 65; //go from ASCII to alphabetical
                    p[i] = (p[i] + k) % 26; //go from origial alphetical letter to new one using key
                    p[i] += 65; //now convert from alphabetical to ASCII
                    printf("%c", p[i]);
                }
                
                else if islower(p[i]) {
                    p[i] -= 97; //go from ASCII to alphabetical
                    p[i] = (p[i] + k) % 26; //go from origial alphetical letter to new one using key
                    p[i] += 97; //now convert from alphabetical to ASCII
                    printf("%c", p[i]);
                    
                }
                //printf("%c%i", p[i]);
            }
            
            //else if it is not an alphabetical character, it will remain unchanged
            else {
                printf("%c", p[i]);
            }
        }
       
       //print a new line at the end of the program and return a 1 to indicate success
       printf("\n");
       return 0; 
    }
}

File: pset6/caesar.py
# encryption program which takes an int (key) from the command line, prompts
# the user for a secret message (aka plaintext) and returns the ciphertext

import cs50
import sys

# get the key
sys.argv[1]

key = int(sys.argv[1])

# if 2 command line arguments are not provided,
# print an error and exit the program
if (len(sys.argv) != 2):
    print("Error: You must provide 2 command line arguments.")
    exit(1)

# prompt the user for the secret message (aka plaintext)
print("What is the secret message?")
plaintext = cs50.get_string()

print("plaintext: ", end="")
print(plaintext)

# print "ciphertext" followed by the actual ciphertext
# which is created using the remaining code
print("ciphertext: ", end="")

# encipher the plaintext:
# for each character in the plaintext string
# if it is alphabetic, shift character by the key
# while preserving the case: caps stay caps,
# lower stay lower, symbols remain unchanged 

for i in plaintext:
    
    # rotate alphabetical characters by key positions
    
    # check to see if the character is a letter 
    if i.isalpha():
        
        # check to see if the letter is uppercase
        # if so, rotate the ASCII char by 65 and print it as a string
        if i.isupper():
            print(chr(((ord(i) - 65 + key) % 26) + 65), end="")
    
        # else if the letter is lowercase,
        # if so, rotate the ASCII char by 97 and print it as a string
        elif i.islower():
            print(chr(((ord(i) - 97 + key) % 26) + 97), end="")
    
    else:
        print(i, end="")

# print a new row and exit the program, indicating success    
print()
exit(0)
    
